### PR TITLE
refactor: change `RdbEngineStrategy` from `abstract class` to `interface`

### PR DESCRIPTION
**Pull request by laysakura**
## Major changes

- change `abstract class RdbEngineStrategy` into `interface RdbEngineStrategy`
- move factory methods of `RdbEngineStrategy` from `RdbEngineStrategy.create()` to `RdbEngineFactory.create()`
- move `QueryUtils.enclose()` and `QueryUtils.enclosedFullTableName()` into implementations of `RdbEngineStrategy`

### CODE REVIEW

File: core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java [hunk 1]
```diff
@@ -3,117 +3,76 @@
...
+   *
+   * @param name The target name to enclose
+   * @return An enclosed string of the target name
+   */
+  String enclose(String name);
 
-  protected String encloseFullTableName(String schema, String table) {
-    return enclose(schema) + "." + enclose(table);
-  }
+  String encloseFullTableName(String schema, String table);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java:77]: It looks like we can use a `default` method here like:
```suggestion
  default String encloseFullTableName(String schema, String table) {
    return enclose(schema) + "." + enclose(table);
  }
```

**Review by brfrn169 — COMMENTED** [review]
LGTM overall. I left several minor comments. Please take a look when you have time!

**Review by komamitsu — APPROVED** [review]
LGTM, thanks!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM except for Toshi's suggestions! Thank you!

File: core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java (hunk 1 above)
Comment (laysakura) [reply core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java:77]: Wow, that's what I want!

Comment (laysakura) [reply core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java]: 421df1b9e1a8430874e7a716b82face68edd39b0 removes duplicate methods in implementations.

File: core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineFactory.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.scalar.db.storage.jdbc;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/** Factory class of subclasses of {@link RdbEngineStrategy} */
+public class RdbEngineFactory {
```
[the remark this answers is not in the document]
Comment (laysakura) [reply core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineFactory.java]: 0e98db6a4974b2d47d6c8ca6eff6d537d23c2718

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**